UnityProject/Assets/Scripts/Items/FloorHazard.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Mirror;
+using Items;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using Random = UnityEngine.Random;
+
+public class FloorHazard : NetworkBehaviour {
+    public DamageType damageType;
+    public AttackType attackType;
+    public float damage;
+    public bool ignoresShoes = false;
+    public bool ignoresWalking = false;
+    public bool requiresArming = false;
+    public bool armed = false;
+    public float chancePercent = 100f;
+    private BodyPartType[] legs = {BodyPartType.RightLeg,BodyPartType.LeftLeg};
+
+    public void ArmHazard(){
+        armed = true;
+    }
+
+    public void DisarmHazard(){
+        armed = false;
+    }
+
+    public void OnEnterableEnter(BaseEventData eventData)
+	{
+        if(!requiresArming || armed){
```

Comment:
extract and encapsulate this logic into different methods. Too much nesting and complexity on this method alone.